### PR TITLE
Fix random values regenerating

### PR DESCRIPTION
**Pull request by Kajot-dev**
- Fixes regeneration of random secrets
- Adds support for existingSecret for `swift` and `oss`

Fixes: #1609 #1549

Essentially does the same thing as #1550 but with different approach

### CODE REVIEW

File: templates/registry/registry-dpl.yaml [hunk 1]
```diff
@@ -120,6 +120,33 @@ spec:
...
+              optional: true
+        {{- end }}
+        {{- if .Values.persistence.imageChartStorage.oss.existingSecret }}
+        - name: REGISTRY_STORAGE_OSS_ACCESSKEYSECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.persistence.imageChartStorage.oss.existingSecret }}
+              key: REGISTRY_STORAGE_OSS_ACCESSKEYSECRET
+              optional: true
+        {{- end}}
```
Comment (zyyw) [inline templates/registry/registry-dpl.yaml:149]: We may also need to add this code block to `registryctl` container within `registry.dpl` deployment.

Comment (Kajot-dev) [reply templates/registry/registry-dpl.yaml:149]: @zyyw You're correct. Fixed!

**Comment by Kajot-dev** [comment]
@zyyw Reverted generating internal database password as this could possibly break existing installations and changes current behavior which allows empty password for internal database.
Also resolved conflicts.

File: values.yaml [hunk 2]
```diff
@@ -291,6 +291,8 @@ persistence:
       username: username
       password: password
       container: containername
+      # keys in existing secret must be REGISTRY_STORAGE_SWIFT_PASSWORD, REGISTRY_STORAGE_SWIFT_SECRETKEY, REGISTRY_STORAGE_SWIFT_ACCESSKEY
```
Comment (zyyw) [inline values.yaml:294]: Please add an entry in `README.md` for this item.

**Review by zyyw — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Comment by kevin7s-io** [comment]
@OrlinVasilev,

Seeing the same in ArgoCD. It's not clear to me why the checksum is changing as diffing the ``{release}-harbor-core`` secret shows no differences between sync runs.

FWIW, I ended up just ignoring the checksum annotations in the ArgoCD app (you can be more tactical than this -- I'm lazy):

```yaml
  ignoreDifferences:
    - group: "*"
      kind: Secret
      jsonPointers:
        - /data
    - group: apps
      kind: Deployment
      jsonPointers:
        - /spec/template/metadata/annotations
```

**Pull request merged**